src/connectionconfig/connectionDialogWebviewController.ts

```diff
@@ -307,53 +308,22 @@ export class ConnectionDialogWebviewController extends FormWebviewController<
         });
 
         this.registerReducer("createConnectionGroup", async (state, payload) => {
-            const addedGroup = createConnectionGroupFromSpec(payload.connectionGroupSpec);
-
-            try {
-                await this._mainController.connectionManager.connectionStore.connectionConfig.addGroup(
-                    addedGroup,
-                );
-                sendActionEvent(
-                    TelemetryViews.ConnectionDialog,
-                    TelemetryActions.SaveConnectionGroup,
-                    { newOrEdit: "new" },
-                );
-            } catch (err) {
-                state.formError = getErrorMessage(err);
-                sendErrorEvent(
-                    TelemetryViews.ConnectionDialog,
-                    TelemetryActions.SaveConnectionGroup,
-                    err,
-                    false, // includeErrorMessage
-                    undefined, // errorCode
-                    err.Name, // errorType
-                    {
-                        failure: err.Name,
-                    },
-                );
-            }
-
-            sendActionEvent(TelemetryViews.ConnectionDialog, TelemetryActions.SaveConnectionGroup);
-
-            state.dialog = undefined;
-
-            state.formComponents.groupId.options = await this.getConnectionGroups();
-            state.connectionProfile.groupId = addedGroup.id;
-            state.connectionGroups =
-                await this._mainController.connectionManager.connectionStore.connectionConfig.getGroups();
+            const updatedState = (await createConnectionGroup(
```

Comment:
Unsure about the coupling between ConnectionDialogWebviewState and ConnectionGroupWebviewController.  Can we refactor this to return both addedGroup and formError and keep all the state updates in the respective controllers?